test/infrastructure/docker/api/v1alpha3/conversion.go

```diff
@@ -208,3 +232,85 @@ func Convert_v1alpha3_Condition_To_v1_Condition(in *clusterv1alpha3.Condition, o
 func Convert_v1_Condition_To_v1alpha3_Condition(in *metav1.Condition, out *clusterv1alpha3.Condition, s apiconversion.Scope) error {
 	return clusterv1alpha3.Convert_v1_Condition_To_v1alpha3_Condition(in, out, s)
 }
+
+func Convert_v1alpha3_DockerMachineStatus_To_v1beta2_DockerMachineStatus(in *DockerMachineStatus, out *infrav1.DockerMachineStatus, s apiconversion.Scope) error {
+	if err := autoConvert_v1alpha3_DockerMachineStatus_To_v1beta2_DockerMachineStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// Reset conditions from autogenerated conversions
+	// NOTE: v1alpha3 conditions should not be automatically be converted into v1beta2 conditions.
+	out.Conditions = nil
+
+	if in.Conditions != nil {
+		out.Deprecated = &infrav1.DockerMachineDeprecatedStatus{}
+		out.Deprecated.V1Beta1 = &infrav1.DockerMachineV1Beta1DeprecatedStatus{}
+		clusterv1alpha3.Convert_v1alpha3_Conditions_To_v1beta2_Deprecated_V1Beta1_Conditions(&in.Conditions, &out.Deprecated.V1Beta1.Conditions)
+	}
+
+	if in.Ready {
+		if out.Initialization == nil {
+			out.Initialization = &infrav1.DockerMachineInitializationStatus{}
+		}
+		out.Initialization.Provisioned = in.Ready
+	}
+
+	return nil
+}
+
+func Convert_v1alpha3_DockerClusterStatus_To_v1beta2_DockerClusterStatus(in *DockerClusterStatus, out *infrav1.DockerClusterStatus, s apiconversion.Scope) error {
+	// NOTE: custom conversion func is required because status.conditions has been added in v1beta2.
+	if err := autoConvert_v1alpha3_DockerClusterStatus_To_v1beta2_DockerClusterStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// Reset conditions from autogenerated conversions
+	// NOTE: v1alpha3 conditions should not be automatically be converted into v1beta2 conditions.
+	out.Conditions = nil
+	if in.Conditions != nil {
+		out.Deprecated = &infrav1.DockerClusterDeprecatedStatus{}
+		out.Deprecated.V1Beta1 = &infrav1.DockerClusterV1Beta1DeprecatedStatus{}
+		clusterv1alpha3.Convert_v1alpha3_Conditions_To_v1beta2_Deprecated_V1Beta1_Conditions(&in.Conditions, &out.Deprecated.V1Beta1.Conditions)
+	}
+
+	out.Initialization = nil
```

Comment:
Q: Why do we have to set this to nil here?

Reply:
It's mistake. Thanks.